representing-data/practice-representing-data.md

```diff
@@ -1,92 +1,538 @@
+        self.assertEqual(data["dog_count"], 10)
+        self.assertEqual(data["cat_count"], 0)
+        self.assertEqual(data["pigeon_count"], 31)
 ```
+### !end-tests
+### !explanation
 
-Create a variable named `groceries` and assign it to a list that contains your own grocery list. Run this code to make sure you don't have any syntax errors or runtime errors.
+An example of a working implementation:
 
 ```python
+def get_friend_data():
+    data = {
+        "date": "2010-06-01",
+        "location": "park",
+        "crow_count": 3,
+        "dog_count": 10,
+        "cat_count": 0,
+        "pigeon_count": 31
+    }
+    return data
+```
+
+### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 8 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: KkABtU
+* title: Representing Data
+### !question
+
+| Name   | Age |
+| ------ | --- |
+| Mae    | 31  |
+| Brandi | 22  |
+| Dane   | 44  |
+| Neo    | 39  |
+| Amara  | 36  |
+| Jules  | 28  |
+
+Using the data above, create a list of contestants' names. The names should be in the order of descending age, where the oldest contestant's name is first, and the youngest contestant's name is last.
 
-# Make a variable named groceries in the line above this comment
+Put that list in the function `sort_contestants_age_desc()` as a list literal and return it.
 
-print(f"My grocery list includes {groceries}")
+### !end-question
+### !placeholder
+
+```python
+def sort_contestants_age_desc():
+    pass
+```
+### !end-placeholder
+### !tests
+```python
+import unittest
+from main import *
+
+class TestPython1(unittest.TestCase):
+    def test_sort_contestants_age_desc(self):
+        contestants = sort_contestants_age_desc()
+
+        self.assertEqual(
+            contestants, ["Dane", "Neo", "Amara", "Mae", "Jules", "Brandi"])
 ```
+### !end-tests
+### !explanation
 
-Here is a function named `add_grocery(grocery_list, new_item)`. Inside of the function body `add_grocery`, add code that will append the `new_item` to `grocery_list`.
+An example of a working implementation:
 
 ```python
-def add_grocery(grocery_list, new_item):
+def sort_contestants_age_desc():
+    contestants = [
+        "Dane", "Neo", "Amara", "Mae", "Jules", "Brandi"
+    ]
+    return contestants
+```
+### !end-explanation
+### !end-challenge
+
+<!-- Question 9 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: rmb4AO
+* title: Representing Data
+### !question
+
+| Field                    | Expected Data Type | Value |
+| ------------------------ | ------------------ | ----- |
+| name                     | string             | -     |
+| is_morning_person        | boolean            | -     |
+| is_night_owl             | boolean            | -     |
+| fav_song_title           | string or `None`   | -     |
+| num_of_household_members | integer            | -     |
```

Comment:
Don't need the hyphens?
```suggestion
| Field                    | Expected Data Type | Value |
| ------------------------ | ------------------ | ----- |
| name                     | string             |       |
| is_morning_person        | boolean            |       |
| is_night_owl             | boolean            |       |
| fav_song_title           | string or `None`   |       |
| num_of_household_members | integer            |       |
```

Reply:
Thanks! Bad habit since I used to have a markdown preview tool that (incorrectly) broke on empty/white-space table cells